lib/Twig/Environment.php

```diff
@@ -1067,11 +1079,11 @@ public function addGlobal($name, $value)
                 $this->globals = $this->initGlobals();
             }
 
-            /* This condition must be uncommented in Twig 2.0
             if (!array_key_exists($name, $this->globals)) {
-                throw new LogicException(sprintf('Unable to add global "%s" as the runtime or the extensions have already been initialized.', $name));
+                // The deprecation notice must be turned into the following exception in Twig 2.0
+                @trigger_error(sprintf('Registering global variable "%s" at runtime or when the extensions have already been initialized is deprecated.', $name), E_USER_DEPRECATED);
+                //throw new LogicException(sprintf('Unable to add global "%s" as the runtime or the extensions have already been initialized.', $name));
```

Comment:
@Taluu the deprecation does not prevent it from working in 1.x. We are deprecating this possibility but not breaking it.
2.0 will forbid it, which will allow to simplify some code (as we will have the list of globals known without allowing to add new ones)